src/App.tsx

```diff
@@ -1,223 +1,157 @@
     }
   }, []);
 
+  const userId = user ? user.id : 0;
+
+  const addTodo = async (todoData: Omit<Todo, 'id'>) => {
+    try {
+      const tempTodo = {
+        ...todoData,
+        id: 0,
+      };
+
+      setTemporaryTodo(tempTodo);
+
+      const newTodo = await creatTodo(todoData);
+
+      setTodos(currentTodoArray => [...currentTodoArray, newTodo]);
+      setTemporaryTodo(null);
+    } catch {
+      showErrorMessage('Unable to add a todo');
+      setTemporaryTodo(null);
+    }
+  };
+
+  const deleteTodoFromData = async (todoId: number) => {
+    try {
+      setDeleteTodoIdFromArray(prevList => [...prevList, todoId]);
+
+      const deletedTodo = await deleteTodo(todoId);
+
+      setTodos(oldAttay => oldAttay.filter(todo => todo.id !== todoId));
+
+      setDeleteTodoIdFromArray(prevList => (
+        prevList.filter(id => id !== todoId)
+      ));
+
+      return deletedTodo;
+    } catch {
+      setDeleteTodoIdFromArray(prevList => (
+        prevList.filter(id => id !== todoId)
+      ));
+      showErrorMessage('Unable to delete a todo');
+
+      return false;
+    }
+  };
+
+  const todoFilteredCompleted = useCallback((value: FilterTodoComplete) => {
+    switch (value) {
+      case FilterTodoComplete.Active:
+        return todos.filter(todo => todo.completed === false);
+
+      case FilterTodoComplete.Completed:
+        return todos.filter(todo => todo.completed === true);
+
+      default:
+        return todos;
+    }
+  }, [todos]);
+
+  const visibleTodo = todoFilteredCompleted(todoToComplete);
+
+  const todoFilteredStatusCompleted = visibleTodo.filter(todo => (
+    todo.completed === true
+  ));
+
+  const deleteTodosStatusCompleted = () => {
+    todos.forEach(todo => {
+      if (todo.completed === true) {
+        deleteTodoFromData(todo.id);
+      }
+    });
+  };
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
+        <TodoHeader
+          onSubmit={addTodo}
+          title={title}
+          setTitle={setTitle}
+          showErrorMessage={showErrorMessage}
+          userId={userId}
+          temporaryTodo={temporaryTodo}
+          deleteTodoFromArray={deleteTodoIdFromArray}
+
+        />
 
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+        {(visibleTodo.length !== 0 || temporaryTodo) && (
+          <>
+            <TodoList
+              todos={visibleTodo}
+              deleteTodoFromData={deleteTodoFromData}
+              temporaryTodo={temporaryTodo}
+              deleteTodo={deleteTodoIdFromArray}
```

Comment:
deleteTodo={deleteTodoIdFromArray}
it's a little bit confusing, make it more understandable